src/System.Memory/tests/BuffersExtensions/BuffersExtensions.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+
+namespace System.Buffers.Tests
+{
+    public class BuffersExtensionsTests
+    {
+        [Fact]
+        public void WritingToSingleSegmentBuffer()
+        {
+            IBufferWriter<byte> bufferWriter = new TestBufferWriterSingleSegment();
+            bufferWriter.Write(Encoding.UTF8.GetBytes("Hello"));
+            bufferWriter.Write(Encoding.UTF8.GetBytes(" World!"));
+            Assert.Equal("Hello World!", bufferWriter.ToString());
+        }
+
+        [Fact]
+        public void WritingToMultiSegmentBuffer()
+        {
+            IBufferWriter<byte> bufferWriter = new TestBufferWriterMultiSegment();
+            bufferWriter.Write(Encoding.UTF8.GetBytes("Hello"));
+            bufferWriter.Write(Encoding.UTF8.GetBytes(" World!"));
+            Assert.Equal("Hello World!", bufferWriter.ToString());
+        }
+
+        private class TestBufferWriterSingleSegment : IBufferWriter<byte>
+        {
+            byte[] _buffer = new byte[1000];
+            int _written = 0;
+
+            public Span<byte> GetBuffer => _buffer;
+
+            public void Advance(int bytes)
+            {
+                _written += bytes;
+            }
+
+            public Memory<byte> GetMemory(int minimumLength = 0)
+            {
+                return ((Memory<byte>)_buffer).Slice(_written);
+            }
+
+            public Span<byte> GetSpan(int minimumLength)
+            {
+                return ((Span<byte>)_buffer).Slice(_written);
+            }
+
+            public int MaxBufferSize { get; } = Int32.MaxValue;
+
+            public override string ToString()
+            {
+                return Encoding.UTF8.GetString(_buffer.AsSpan(0, _written));
```

Comment:
This API only exists on .NET Core 2.1 and not on other platforms like netfx.
https://apisof.net/catalog/System.Text.Encoding.GetString(ReadOnlySpan%3CByte%3E)

Consider using the array based API or add an #ifdef for netfx.
https://apisof.net/catalog/System.Text.Encoding.GetString(Byte())